Client/Frontend/InternalSchemeHandler/InternalSchemeHandler.swift

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import WebKit
+import Shared
+
+enum InternalPageSchemeHandlerError: Error {
+    case badURL
+    case noResponder
+    case responderUnableToHandle
+}
+
+protocol InternalSchemeResponse {
+    func response(forRequest: URLRequest) -> (URLResponse, Data)?
+}
+
+private var downloadTasks = WeakList<WKURLSchemeTask>()
+
+class InternalSchemeHandler: NSObject, WKURLSchemeHandler {
+
+    static func response(forUrl url: URL) -> URLResponse {
+        return URLResponse(url: url, mimeType: "text/html", expectedContentLength: -1, textEncodingName: "utf-8")
+    }
+
+    // Responders are looked up based on the path component, for instance responder["about/license"] is used for 'internal://local/about/license'
+    static var responders = [String: InternalSchemeResponse]()
+
+    // Unprivileged internal:// urls are probably resources on the page. Change to https:// and try download them.
```

Comment:
nit: A better comment 'Change to https:// and try download them or try to load the resource from the app bundle'.